src/common/host_device_vector.h

```diff
@@ -191,6 +193,11 @@ inline GPUAccess operator-(GPUAccess a, GPUAccess b) {
   return static_cast<GPUAccess>(static_cast<int>(a) - static_cast<int>(b));
 }
 
+enum GPUDataPreservation {
```

Comment:
This looks like `bool`.

Reply:
Yeah, it's the last suggestion in https://abseil.io/tips/94, but looks like using `bool` also works.